File: Arrays/Easy/977/977.c
#include <stdio.h>
#include <stdlib.h>

int* sortedSquares(int* nums, int numsSize, int* returnSize){
    *returnSize = numsSize;
    int *result = (int *)malloc(sizeof(int) * numsSize);
    int left = 0, right = numsSize - 1, i = right;
    while(left <= right){
        if(abs(nums[left]) >= abs(nums[right])){
            result[i--] = nums[left] * nums[left];
            left++;
        }else{
            result[i--] = nums[right] * nums[right];
            right--;
        }
    }
    return result;
}

File: Arrays/Easy/977/977.js
var sortedSquares = function(nums) {
    let arr = new Array(nums.length)
    let left = 0
    let right = nums.length - 1
    let i = right
    while(left <= right){
        if(Math.abs(nums[left]) >= Math.abs(nums[right])){
            arr[i--] = nums[left] * nums[left++]
        }else{
            arr[i--] = nums[right] * nums[right--]
        }
    }
    return arr
};